### PR TITLE
Fixes area portals including each other's target turfs.

### PR DESCRIPTION
**Pull request by ghost**
[bugfix]
:cl:
 * bugfix: Area portals no longer include each other as target turfs if the offset lands on another.

### CODE REVIEW

File: code/game/turfs/portal.dm [hunk 1]
```diff
@@ -33,7 +33,10 @@
 /turf/portal/update_icon()
 	overlays.Cut()
 	vis_contents.Cut()
-	target_turf = locate(src.x+teleport_x,src.y+teleport_y,src.z+teleport_z)
+	var/turf/temp_turf = locate(src.x+teleport_x,src.y+teleport_y,src.z+teleport_z)
+	if(istype(temp_turf,/turf/portal))
+		return
```
Comment (Exxion) [inline code/game/turfs/portal.dm:39]: It should probably emit a warning rather than failing silently

Comment (ghost) [reply code/game/turfs/portal.dm:39]: Warning then fail or just warning?

Comment (Exxion) [reply code/game/turfs/portal.dm:39]: Warning then fail. Still don't want this to happen, it would just be better if it notified someone of the mistake.

Comment (ghost) [reply code/game/turfs/portal.dm:39]: Okay, done

**Comment by GreatBigFailure** [comment]
Infinitemotion
You took this away from me

**Comment by Exxion** [comment]
Where's the area portal in that?

**Pull request merged**